py/farm_ng/core/event_client.py

```diff
@@ -251,8 +255,12 @@ async def request_reply(
             return RequestReplyReply()
 
         # get the current count and increment it
-        count: int = self._counts.get(path, 0)
-        self._counts[path] = count + 1
+        path_received = f"{path}/receive_count"
+        if path_received not in self._metrics.data:
+            self._metrics.data[path_received] = 0
+
+        count: int = int(self._metrics.data[path_received])
+        self._metrics.data[path_received] = count + 1
```

Comment:
Suggest to look at the latency between `SERVICE_SEND` & `CLIENT_RECEIVE` stamps (in host monotonic clock) here and especially in the subscribe method. This has proven to be a very important metric and will be a quick indicator of issues in our pipeline.

This will not be meaningful if service and client are on a different host (e.g. using a dev station as client) so a first version could only report IFF the service and client are on the same host. And we could add a host/client monotonic clock synchronization to the API in the future.

Reply:
This is really request /reply metrics in this location.  Above in subscirbe we can compute the latency between SERVICE_SEND and CLIENT_RECEIVE (as they are in the timestamps field) and this is the metric we are quite interested in, in addition to count and frequency.